contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -212,7 +213,8 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         uint256 totalRewards,
         uint8 distributionType,
         uint32 startTime,
-        uint32 endTime
+        uint32 endTime,
+        uint32 halfLifeInDays
```

Comment:
Since it's also irrelevant for flat distribution, I'd suggest stopping using a generic single `createProgram` function and splitting it into two specific functions with their own parameters and validation (and sharing the common logic via another `private` function)

Reply:
done